src/clustering/administration/main/command_line.cc

```diff
@@ -443,6 +447,28 @@ boost::optional<int> parse_join_delay_secs_option(
     }
 }
 
+boost::optional<int> parse_node_reconnect_timeout_secs_option(
+        const std::map<std::string, options::values_t> &opts) {
+    if (exists_option(opts, "--cluster-reconnect-timeout")) {
+        const std::string timeout_opt = get_single_option(opts, "--cluster-reconnect-timeout");
+        uint64_t node_reconnect_timeout_secs;
+        if (!strtou64_strict(timeout_opt, 10, &node_reconnect_timeout_secs)) {
+            throw std::runtime_error(strprintf(
+                    "ERROR: cluster-reconnect-timeout should be a number, got '%s'",
+                    timeout_opt.c_str()));
+        }
+        if (node_reconnect_timeout_secs > std::numeric_limits<int>::max() ||
+            node_reconnect_timeout_secs * 1000 > std::numeric_limits<int>::max()) {
+            throw std::runtime_error(strprintf(
+                    "ERROR: cluster-reconnect-timeout is too large. Must be at most %d",
+                    std::numeric_limits<int>::max()));
```

Comment:
I think the maximum in the error message is still consistently the actual maximum (numeric_limits<int>::max === biggest supported integer right?). So the error message is correct, it was the checks that needed to ensure that:

  a) the input number is indeed within that threshold
  b) secondarily, since we intend to use it as ms the number multiplied by 1000 should still be within that range

Perhaps you think the error message should be more descriptive in the event that someone enters a number that is too big only once we are testing for conversion to ms?